### PR TITLE
add unit tests for SerDe of commit messages and contents

### LINKED ISSUE #1093: More complete unit tests of TableCommitMetaStoreWorker
As per #1023 we have to add a more comlete set of tests on `TableCommitMetaStoreWorker`. Particularly the serde ops and any edge cases w/ the serde

### PR DESCRIPTION
**Pull request by rymurr**
fixes #1093

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1113)

### CODE REVIEW

File: model/src/main/java/org/projectnessie/model/HiveTable.java [hunk 1]
```diff
@@ -33,4 +36,50 @@
...
   public abstract List<byte[]> getPartitions();
 
+  /**
+   * Because of the List of byte arrays Immutables doesn't generate a correct hashcode.
+   *
+   * <p> We hand compute the List elements to ensure the byte arrays hash codes are consistent.
+   */
+  @Override
+  public int hashCode() {
+    int h = 5381;
```
Comment (snazy) [inline model/src/main/java/org/projectnessie/model/HiveTable.java:46]: I'd prefer the "standard" way common in the JDK. I.e. initialize with `1` and use `h = (31 * h) + xyz`, like `java.util.Arrays#hashCode(java.lang.Object[])` does.

**Review by snazy — CHANGES_REQUESTED** [review]

File: model/src/main/java/org/projectnessie/model/HiveTable.java (hunk 1 above)
Comment (rymurr) [reply model/src/main/java/org/projectnessie/model/HiveTable.java]: cool, fixed. That was a direct copy from immutables but I agree, we should stick to the standard

**Review by snazy — APPROVED** [review]

**Review by laurentgo — APPROVED** [review]
